python/cudf/cudf/core/index.py

```diff
@@ -776,6 +765,66 @@ def _split(self, splits):
     def _binaryop(self, other, op: str):
         return self._as_int64()._binaryop(other, op=op)
 
+    @property  # type: ignore
+    @_cudf_nvtx_annotate
+    def _column(self):
+        return self._as_int64()._column
+
+    @property  # type: ignore
+    @_cudf_nvtx_annotate
+    def _columns(self):
+        return self._as_int64()._columns
+
+    @property  # type: ignore
+    @_cudf_nvtx_annotate
+    def values_host(self):
+        return self.to_pandas().values
+
+    @_cudf_nvtx_annotate
+    def argsort(
+        self,
+        ascending=True,
+        na_position="last",
+    ):
+        if na_position not in {"first", "last"}:
+            raise ValueError(f"invalid na_position: {na_position}")
+
+        indices = cupy.arange(0, len(self))
+        if (ascending and self._step < 0) or (
+            not ascending and self._step > 0
+        ):
+            indices = indices[::-1]
+        return indices
+
+    def where(self, cond, other=None, inplace=False):
+        return self._as_int64().where(cond, other, inplace)
+
+    def to_numpy(self):
+        return self._as_int64().to_numpy()
+
+    def replace(self, to_replace, value):
+        return self._as_int64().replace(to_replace=to_replace, value=value)
+
+    def to_arrow(self):
+        return self._values.to_arrow()
+
+    def __array__(self, dtype=None):
+        raise TypeError(
+            "Implicit conversion to a host NumPy array via __array__ is not "
+            "allowed, To explicitly construct a GPU matrix, consider using "
+            ".to_cupy()\nTo explicitly construct a host matrix, consider "
+            "using .to_numpy()."
+        )
+
+    def nunique(self):
+        return self._stop  # or len(self)
```

Comment:
Does `len(self)` always equal `self._stop`? What happens if you test it on something like `cudf.RangeIndex(9, 10)`?

Reply:
I'm guessing that this method is being called indirectly through some other test, but is not being tested directly so I would recommend adding that.